app/src/main/java/net/c306/photopress/ui/welcome/SelectBlogViewModel.kt

```diff
@@ -23,41 +24,48 @@ import javax.inject.Inject
 internal class SelectBlogViewModel @Inject constructor(
     private val wpService: WpService,
     private val authPrefs: AuthPrefs,
-    private val settings: Settings,
+    settings: Settings,
 ) : ViewModel() {
 
     private val _blogList = MutableLiveData<List<Blog>?>()
     val blogList: LiveData<List<Blog>?> = _blogList
 
-    private val selectedBlog = MutableStateFlow<Blog?>(null)
+    val isBlogSelected = settings.selectedBlogId
+        .mapLatest { it > -1 }
+        .stateIn(
+            scope = viewModelScope,
+            started = SharingStarted.WhileSubscribed(stopTimeoutMillis = 3000),
+            initialValue = false,
+        )
+
+    val selectBlogWelcomeSubtitle = settings.selectedBlogId
```

Comment:
We just observe selected blog from the repository, instead of depending on the fragment to inform us.